code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -412,7 +420,14 @@ There are several things that need to be remembered:
 			handled_by_bodytype = FALSE
 			icon_file = 'icons/mob/clothing/head.dmi'
 
-		head_overlay = head.build_worn_icon(default_layer = HEAD_LAYER, default_icon_file = icon_file)
+		// SKYRAT EDIT ADDITION
+		var/species_override_icon
+		if(dna.species.bodytype & BODYTYPE_SNOUTED)
+			if(worn_item.supports_variations_flags & CLOTHING_SNOUTED_VARIATION)
+				species_override_icon = wear_mask.worn_icon_muzzled || SNOUTED_HEAD_FILE
+		// SKYRAT EDIT END
+
+		head_overlay = head.build_worn_icon(default_layer = HEAD_LAYER, default_icon_file = icon_file, override_file = species_override_icon) // SKYRAT EDIT CHANGE
```

Comment:
this stuff is supposed to come before `icon_exists() since the entire purpose of that is to ensure invisible sprites dont occur